### PR TITLE
Bugfix/spline spark agent 581 integration tests

### LINKED ISSUE #581: Core jar to depend on a fixed Spark version regardless of active Maven profile
`spark-xy` profile should only affect integration tests, not the _core_ jar.
The final production artifacts are always built with the fixed Spark version (2.4), and integration tests should test them as they are. Currently the Spark version is managed by `spark-x.y.` profiles for the entire project. Thus the Spark versions used for the _core_ module compilation and for the integration tests run are always properly aligned, which doesn't reflect the real usage.

### PR DESCRIPTION
**Pull request by wajda**
fixes #581 

1. Add ability to enable/disable any plugin in the configuration
2. Fix spark dependencies accordingly to the issue description

### CODE REVIEW

File: core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala [hunk 1]
```diff
@@ -73,10 +73,22 @@ class AutoDiscoveryPluginRegistry(
     constr.newInstance(args: _*).asInstanceOf[Plugin]
   }
 
+  private def isPluginEnabled(pc: Class[Plugin]): Boolean = {
+    val pluginConf = conf.subset(pc.getName)
+    val isEnabled = pluginConf.getBoolean(EnabledConfProperty, EnabledByDefault)
+    if (!isEnabled) {
+      log.warn("Plugin {} is disabled in the configuration.", pc.getName)
```
Comment (cerveada) [inline core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala]: The message is evaluated lazily already no need for `{}` or is there?

Comment (wajda) [reply core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala]: it's SLF4J. Don't you confuse it with SLF4S? SLF4J is Java API, there's no pass-by-name for method params.

Comment (cerveada) [reply core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala]:
```
  protected def logWarning(msg: => String): Unit = {
    if (log.isWarnEnabled) log.warn(msg)
  }
```

Comment (cerveada) [reply core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala]:
```suggestion
      logWarning(s"Plugin ${pc.getName} is disabled in the configuration.")
```

Comment (wajda) [reply core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala]: Ah, Ok, I can use another method `logWarning` that is Scala one with pass-by-name :)

Comment (wajda) [reply core/src/main/scala/za/co/absa/spline/harvester/plugin/registry/AutoDiscoveryPluginRegistry.scala]: done

**Review by cerveada — DISMISSED** [review]
LGTM

**Pull request merged**